### DISCUSSION TITLE
large sate/service files

### DISCUSSION THREAD

**Discussion by kingjordan** [state-adapt/state-adapt#95] [Q&A] [answered]
just a quick question, looking for feedback.

I have an app with a feature that is quite state heavy.  I am looking to break up my large files to make them more manageable. The largest file is an angular service with 30+ sources that get linked to a store adapter as well as "effect" observables that trigger data persistence auth etc.  The adapter and initial state are already in their own file.  I prefer to keep code co located so breaking files into actions(sources) / effects(observable that listen to sources and trigger database etc) / reducers (adapters) does not seem ideal but is maybe the best way.

Anyone else have to deal with growing state files?  What did you do to keep things organized well.

Cheers

**Comment by mfp22** [comment, marked as answer]
[`joinAdapters`](https://state-adapt.github.io/docs/core#joinadapters) might be able to help. 

After that, separating the adapter to its own file is common. 

Or if the states are fairly unrelated, you break it into multiple stores. If you need the state from one while calculating a state change in another, you can include the first state as context for source updates so it gets included in the payload for the other adapter. 

I would have to see an example for any other ideas, but there's usually not more to do beyond that.

**Reply by kingjordan** [reply]
thanks mike!
